csrc/ir/builder.cpp

```diff
@@ -373,12 +378,17 @@ Val* SimplifyingIrBuilder::divExpr(Val* lhs, Val* rhs) {
   if (rhs->isOneInt()) {
     return lhs;
   }
+  if (lhs->sameAs(rhs)) {
+    return lhs->fusion()->oneVal(lhs->dtype());
+  }
   return IrBuilder::divExpr(lhs, rhs);
 }
 
 Val* SimplifyingIrBuilder::ceilDivExpr(Val* lhs, Val* rhs) {
   if (rhs->isOneInt()) {
     return lhs;
+  } else if (lhs->sameAs(rhs)) {
```

Comment:
What if rhs is zero or inf or nan?

Reply:
Guarding with `isIntegralScalar` handles inf and nan, but I don't have a solution for zero. Leaving as is, we might replace `ceilDiv(0, 0)` which should raise an exception with `1`.  I will remove this branch.